keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/ReshapeVertex.java

```diff
@@ -26,8 +29,15 @@ protected DualNumber dualOp(DualNumber dualNumber) {
 
     @Override
     public Map<Vertex, PartialDerivatives> reverseModeAutoDifferentiation(PartialDerivatives derivativeOfOutputsWithRespectToSelf) {
-        //TODO
-        throw new UnsupportedOperationException("Reverse mode autodiff not supported for this operation");
+        Map<Vertex, PartialDerivatives> reshapedDerivatives = new HashMap<>();
+
+        for (Map.Entry<VertexId, DoubleTensor> partialDerivative : derivativeOfOutputsWithRespectToSelf.asMap().entrySet()) {
+            int[] newPartialShape = TensorShape.concat(getShape(), inputVertex.getShape());
```

Comment:
I think this only works when taking the derivative or this reshape wrt to something else.